hubblestack/files/hubblestack_nova/oval_scanner.py

```diff
@@ -166,23 +174,128 @@ def build_impact_report(impact, report={}):
     return report
 
 
-def get_impact(local_ver, name, ver, title, cve, advisory, severity):
-    """Compare local package ver to vulnerability ver in rpm distros"""
-    logging.debug('get_rpm_impact')
+def determine_impact(vulns, local_pkgs, distro_name, queue):
+    """Determine impacts based on pkg comparisons"""
+    threads = []
+    for data in vulns.values():
+        for pkg in data['pkg']:
+            name = pkg['name']
+            ver = pkg['version']
+            if name in local_pkgs:
+                title = data['title']
+                cve = data['cve']
+                if 'severity' in data:
+                    severity = data['severity']
+                else:
+                    severity = 'N/A'
+                if distro_name in ('centos', 'redhat'):
+                    advisory = data['rhsa']
+                elif 'advisories' in data:
+                    advisory = data['advisories']
+                else:
+                    advisory = cve
+                worker = threading.Thread(target=functools.partial(
```

Comment:
Use a thread-pool. (Read below comment for threads)

Reply:
Do not use threads at all.